primes.py

```diff
@@ -40,7 +40,7 @@ def __getitem__(self, key: Any) -> Any:
 
         num_required = max((key.start or -1)+1, key.stop or 0) if isinstance(key, slice) else key+1
         if len(self) < num_required:
-            self.primes += n_primes(num_required, self)[len(self):]
+            self.primes = n_primes(num_required, self)
```

Comment:
Thinking with this is that it'd reduce the amount of copying required (so be O(1) rather than O(num_required - len(self)), although thinking about it it needs a `list` call anyway, which will make it O(len(list))

https://docs.python.org/3.8/library/stdtypes.html#list

> The constructor builds a list whose items are the same and in the same order as iterable’s items. iterable may be either a sequence, a container that supports iteration, or an iterator object. If iterable is already a list, a copy is made and returned, similar to iterable[:].

Could get around this by returning the underlying `primes` list instead of the `Primes` instance, but I think that is worse, or grab it off the returned value, but those would necessitate changing the return type which would make the library worse